src/media/MediaAnnotator.ts

```diff
@@ -0,0 +1,132 @@
+                manager.destroy();
+                this.managers.delete(manager);
+            }
+        });
+
+        if (this.managers.size === 0) {
+            this.managers.add(new PopupManager({ location: MEDIA_LOCATION_INDEX,   referenceEl, resinTags, targetType : FRAME}));
+            this.managers.add(new DrawingManager({ location: MEDIA_LOCATION_INDEX, referenceEl, resinTags, targetType : FRAME}));
+            this.managers.add(new RegionManager({ location: MEDIA_LOCATION_INDEX, referenceEl, resinTags, targetType : FRAME}));
+            this.managers.add(new RegionCreationManager({ location: MEDIA_LOCATION_INDEX, referenceEl, resinTags, targetType : FRAME}));
+        }
+
+        return this.managers;
+    }
+
+  
+
+    // returns the actual video element
+    getReference(): HTMLVideoElement | null | undefined {
+        return this.annotatedEl?.querySelector('video');
+    }
+
+    handleStore = (): void => {
+        const referenceEl = this.getReference();
+
+        if (!referenceEl) {
+            return;
+        }
+
+        if (getCreatorStatus(this.store.getState()) === CreatorStatus.started) {
+            referenceEl.classList.add(CSS_IS_DRAWING_CLASS);
+        } else {
+            referenceEl.classList.remove(CSS_IS_DRAWING_CLASS);
+        }
+    };
+
+    render(): void {
+        const referenceEl = this.getReference();
+        const rotation = getRotation(this.store.getState()) || 0;
+
+        if (!this.annotatedEl || !referenceEl) {
+            return;
+        }
+
+        this.getManagers(this.annotatedEl, referenceEl).forEach(manager => {
+            manager.style({
+                height: `${referenceEl.offsetHeight}px`,
+                left: `${referenceEl.offsetLeft}px`,
+                top: `${referenceEl.offsetTop}px`,
+                transform: `rotate(${rotation}deg)`,
+                width: `${referenceEl.offsetWidth}px`,
+            });
+            manager.render({
+                intl: this.intl,
+                store: this.store,
+            });
+        });
+
+        this.postRender();
+    }
+
+    // This function gets called by the client. We are using the name scrollToAnnotation to match 
+    // the other annotators as the client calls this function when it wants to show an annoation on the
+    // preview. The only difference is that we are using the video element's currentTime to scroll to the 
+    // annotation instead of the scrollLeft and scrollTop of the parent element.
+    scrollToAnnotation(annotationId: string | null): void {
+        if (!annotationId) {
+            return;
+        }
+
+        const annotation = getAnnotation(this.store.getState(), annotationId);
+        const annotationLocation = annotation?.target.location.value ?? 0
+        const video = this.getReference();
+        if (!annotation || !video || !this.annotatedEl) {
+            return;
+        }
+  
+        video.pause();
+        video.currentTime = annotationLocation / 1000;
```

Comment:
I noticed in the demo you gave yesterday that the video pauses and then navigates to the right time. Is there any way to do both of these actions at the same time?